pkg/cloud/amazon/cli.go

```diff
@@ -61,7 +61,7 @@ func InstallKops() error {
 		return err
 	}
 	clientURL := fmt.Sprintf("https://github.com/kubernetes/kops/releases/download/%s/kops-%s-%s", latestVersion, runtime.GOOS, runtime.GOARCH)
-	fullPath := filepath.Join(binDir, fileName)
+	fullPath := filepath.Join(binDir, binary)
```

Comment:
I think you first would need to call `BinaryWithExtension(binary)` here. Otherwise, there is a discrepancy between the file we are writing/creating here and the file whose existence we check for in `ShouldInstallBinary`.
This effects Windows. On macOS and Linux this would work.